cpp/src/arrow/compute/kernels/util_internal.h

```diff
@@ -23,8 +23,11 @@
 
 #include "arrow/array/util.h"
 #include "arrow/buffer.h"
+#include "arrow/chunked_array.h"
 #include "arrow/compute/kernels/codegen_internal.h"
 #include "arrow/compute/type_fwd.h"
+#include "arrow/record_batch.h"
+#include "arrow/table.h"
```

Comment:
nit: I think we should be able to forward-declare Array/RecordBatch/Table instead of including the headers (or include `arrow/type_fwd.h`), which would cut down on the headers that get (transitively) included.

Reply:
I've already tried to include "arrow/type_fwd.h", but I got a message that says that classes have incomplete types and I can't use their methods.

Reply:
Ah, sorry for the confusion - while, as suggested below, these functions will be moved, what would be done here is: 
1. `#include "arrow/type_fwd.h"` inside `util_internal.h`
2. `#include "arrow/record_batch.h"` etc. inside `util_internal.cc`

The motivation is just not to leak the includes to other users of the header, but of course in the implementation we still need the full include.

Reply:
Oh, I got it. I understand it better now, thanks.

Reply:
Can you remove these includes now? They're not needed anymore.

Reply:
@ArianaVillegas Again, there doesn't seem to be a reason for adding these includes while the code didn't change.